pkg/flowaggregator/s3uploader/s3uploader_test.go

```diff
@@ -108,7 +95,10 @@ func TestCacheRecord(t *testing.T) {
 }
 
 func TestBatchUploadAll(t *testing.T) {
-	mockS3Uploader := &mockS3Uploader{testReader: &bytes.Buffer{}}
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockS3Uploader := s3uploadertesting.NewMockS3UploaderAPI(ctrl)
+	mockS3Uploader.EXPECT().Upload(context.TODO(), gomock.Any(), nil).Return(nil, nil)
```

Comment:
please define `ctx := context.Background()` at the beginning of the test case, and use the `ctx` objects every time you need a Context. I don't like `context.TODO()` in tests because it's not like we ever intend to use a different Context.